compiler/plugins/target/ROCM/builtins/tuning/iree_default_tuning_spec_gfx942.mlir

```diff
@@ -19,6 +19,77 @@ transform.named_sequence @apply_op_config(%op: !transform.any_op {transform.read
 
+transform.named_sequence @apply_attn_op_config(%attention: !transform.any_op {transform.readonly},
+                                                %config: !transform.any_param {transform.readonly},
+                                                %decomposition_config: !transform.any_param {transform.readonly}) {
+  transform.annotate %attention "compilation_info" = %config : !transform.any_op, !transform.any_param
+  transform.annotate %attention "decomposition_config" = %decomposition_config : !transform.any_op, !transform.any_param
+  transform.annotate %attention "__tuning_spec_applied__" : !transform.any_op
+  transform.yield
+}
+
+transform.named_sequence @match_attention_f16(%root: !transform.any_op {transform.readonly})
+  -> !transform.any_op {
+  transform.match.operation_name %root ["iree_linalg_ext.attention"] : !transform.any_op
+  %ins, %outs = transform.iree.match.cast_compatible_dag_from_root %root {
+    ^bb0(%query: tensor<?x?x?x?xf16>,
+         %key: tensor<?x?x?x?xf16>,
+         %value: tensor<?x?x?x?xf16>,
+         %softmax_scale: f16,
+         %out: tensor<?x?x?x?xf16>):
+
+      %attn = iree_linalg_ext.attention {indexing_maps = [
+                                          affine_map<(d0, d1, d2, d3, d4, d5) -> (d0, d1, d2, d4)>,
+                                          affine_map<(d0, d1, d2, d3, d4, d5) -> (d0, d1, d5, d4)>,
+                                          affine_map<(d0, d1, d2, d3, d4, d5) -> (d0, d1, d3, d5)>,
+                                          affine_map<(d0, d1, d2, d3, d4, d5) -> ()>,
+                                          affine_map<(d0, d1, d2, d3, d4, d5) -> (d0, d1, d2, d3)>]}
+        ins(%query, %key, %value, %softmax_scale :
+            tensor<?x?x?x?xf16>, tensor<?x?x?x?xf16>, tensor<?x?x?x?xf16>, f16)
+        outs(%out : tensor<?x?x?x?xf16>){
+          ^bb0(%arg0: f32):
+            iree_linalg_ext.yield %arg0 : f32
+        } -> tensor<?x?x?x?xf16>
+  } : (!transform.any_op) -> (!transform.any_value, !transform.any_value)
+
+  transform.yield %root : !transform.any_op
+}
+
+transform.named_sequence
+@match_attention_2x10x4096x64x64x64_f16(%attention: !transform.any_op {transform.readonly})
+  -> (!transform.any_op, !transform.any_param, !transform.any_param) {
+
+  %matched = transform.include @match_attention_f16 failures(propagate) (%attention)
+    : (!transform.any_op) -> !transform.any_op
+
+  %query = transform.get_operand %attention[0] : (!transform.any_op) -> !transform.any_value
+  %key = transform.get_operand %attention[1] : (!transform.any_op) -> !transform.any_value
+  %value = transform.get_operand %attention[2] : (!transform.any_op) -> !transform.any_value
+
+  transform.iree.match.cast_compatible_type %query = tensor<2x10x4096x64xf16> : !transform.any_value
+  transform.iree.match.cast_compatible_type %key = tensor<2x10x64x64xf16> : !transform.any_value
+  transform.iree.match.cast_compatible_type %value = tensor<2x10x64x64xf16> : !transform.any_value
+
```

Comment:
This lowering spec only promotes K and V tensors to shared memory. It does not promote Q to shared memory. This works, because we use VMFMA instructions for QK matmul which take vector<8xf16>, which means we don't loose out on good global memory reads despite not using shared memory for Q.